classes/article/PublishedArticleDAO.inc.php

```diff
@@ -125,6 +128,81 @@ function getPublishedArticles($issueId) {
 	}
 
 	/**
+	 * Get all published article versions ids
+	 * @param $submissionId int
+	 * @param $contextId int optional
+	 * @param $order string optional
+	 * @return array
+	*/
+	function getPublishedSubmissionRevisionIds($submissionId, $contextId = null, $order = SORT_DIRECTION_DESC){
+		$params = array((int) $submissionId);
+		if ($contextId) {
+			$params[] = (int) $contextId;
+		}
+
+		$sql = 'SELECT DISTINCT ss.submission_revision
+			FROM 	submission_settings ss
+				LEFT JOIN published_submissions ps ON ps.submission_id = ss.submission_id
+				LEFT JOIN submissions s ON ps.submission_id = s.submission_id
+			WHERE 	ss.submission_id = ?'.
+				($contextId ? ' AND s.context_id = ?' : '').'
+				AND ss.setting_name = \'datePublished\'
```

Comment:
could it be that the setting_value is empty i.e. do we have to ensure that it is not?